### DISCUSSION TITLE
Is is possible to build amd64 image only on arm platform?

### DISCUSSION THREAD

**Discussion by k0va1** [basecamp/kamal#549] [Q&A] [answered]
Hi, I'm using Apple Silicon, and I want to build locally and push only amd64 image because I'm using linux/amd64 as a target server. I've tried different approaches, but I couldn't figure out how to achieve it. I don't feel like using a remote build because my laptop is much more performant than the application server with 1 CPU and 2 GB of RAM. Is there any way to make it work out?

**Comment by nickhammond** [comment]
I think this might work but I haven't tried it:

``` yml
builder:
  local:
    arch: amd64 
    multiarch: false
```

**Reply by k0va1** [reply]
Thanks for fast reply, but unfortunately no. It still generates build command like this `docker buildx build --push --platform linux/amd64,linux/arm64 --builder kamal-blog-multiarch -t ....`

**Reply by nickhammond** [reply]
The other thought I had was you could setup a simple local VM via Vagrant and point the builder to that as a remote using the local IP which you can hardcode in Vagrant.

**Comment by bibstha** [comment]
I solved this by switching to colima. You can start an amd64 vm with docker using rosetta (should be faster than qemu).
Make sure your docker desktop is not running. `colima` is a drop in replacement for docker desktop.

```
brew install colima
colima start --arch amd64 --vm-type=vz --vz-rosetta
colima ssh
uname -a
# Linux colima 6.1.55-0-virt #1-Alpine SMP PREEMPT_DYNAMIC Sun, 24 Sep 2023 23:14:02 +0000 x86_64 Linux
```

As you can see, the colima VM is a x86_64 Linux machine. `kamal` uses `docker` that runs through `colima` and the default arch is amd64. Here is my `config/deploy.yml`. 

```
builder:
  multiarch: false
```

That's all I needed in the `builder` section.

**Reply by buncis** [reply]
curious isn't --arch am64 forcing colima to use qemu instead vz, and therefore it can't use rosetta?

**Reply by buncis** [reply]
> curious isn't --arch am64 forcing colima to use qemu instead vz, and therefore it can't use rosetta?

running `colima status` can confirm the virtualization
```
~$ colima status
```                                          
```                                                                                           
INFO[0000] colima is running using macOS Virtualization.Framework 
INFO[0000] arch: x86_64                                 
INFO[0000] runtime: docker                              
INFO[0000] mountType: virtiofs                          
INFO[0000] socket: unix:///Users/buncis/.colima/default/docker.sock 
```

**Comment by anderssvendal** [comment]
Had the same issue and found this thread. Adding the solution that worked for me if others end up here. 

```
builder:
  remote:
    arch: amd64
```

The "remote" will be a buildx instance running locally on both amd64 and arm64.

**Reply by gregjotau** [reply]
This seems to work (Thanks  Anders!): 

<img width="1042" alt="Screenshot 2024-01-23 at 17 08 21" src="https://github.com/basecamp/kamal/assets/13047497/42a28f06-b33e-4c0a-8516-bf5583b19f00">

The first container is before adding: 
```
builder:
  remote:
    arch: amd64
```

Seems to save 10 seconds in my case (about 60-70 seconds for second deploy before, now 50-60)

**Reply by Talha345** [reply]
This works however I had to remove the existing builders from my docker engine to make it work.

**Comment by kaka-ruto** [comment]
I have tried the suggestions above, most notably the last one

```
builder:
  remote:
    arch: amd64
```

But I keep getting this error

```
Running docker image rm --force ghcr.io/name/name_staging:45eea3075e64ec33e842afb99e600f30810bdbb2 on 135.181.251.34
  INFO [6623a9a4] Finished in 0.316 seconds with exit status 0 (successful).
  INFO [930dea80] Running docker pull ghcr.io/name/name_staging:45eea3075e64ec33e842afb99e600f30810bdbb2 on 135.181.251.34
  Finished all in 23.3 seconds
  ERROR (SSHKit::Command::Failed): Exception while executing on host 135.181.251.34: docker exit status: 1
docker stdout: 45eea3075e64ec33e842afb99e600f30810bdbb2: Pulling from name/name_staging
docker stderr: no matching manifest for linux/amd64 in the manifest list entries
```
Anyone have an idea how to fix?

**Comment by djmb** [comment]
For anyone looking to build "remotely" on your localhost, I think this should work:

```yml
builder:
  arch: amd64
  remote: unix:///var/run/docker.sock
  local: false
```

**Comment by minhanh811** [comment]
<img width="542" alt="Screenshot 2025-04-18 at 23 53 39" src="https://github.com/user-attachments/assets/47ac4fd2-9078-4b79-bfda-944579708740" />

Use this one in your Docker Desktop setting. Good luck

**Comment by k0va1** [comment, marked as answer]
I'm using OrbStack instead of Docker for Mac(it just works better for everything) + this config.
```
# since version 2.9.0
registry:
  server: "localhost:5555"

builder:
  arch: amd64
```
Works great.
